presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java

```diff
@@ -179,11 +181,23 @@ private void appendProbeIndex(JoinProbe probe)
         if (previousPosition == position) {
             return;
         }
+        estimatedProbeBlockBytes += getEstimatedProbeRowSize(probe);
+    }
+
+    private int getEstimatedProbeRowSize(JoinProbe probe)
+    {
+        if (estimatedProbeRowSize != -1) {
+            return estimatedProbeRowSize;
+        }
+
+        estimatedProbeRowSize = 0;
```

Comment:
i'd prefer to use a local var and populate the class state (field) only when fully calculated.
(one variable more, but makes code easier to reason about)